docs/grover.rst

```diff
@@ -4,14 +4,103 @@ Grover's Search Algorithm and Amplitude Amplification
 Overview
 --------
 
-Quantum search algorithm and more via amplitude amplification.
+This module implements Grover's Search Algorithm, and the more general Amplitude Amplification
+Algorithm. Grover's Algorithm solves the following problem:
+
+Given a collection of basis states {:math:`\ket{y}_i`}, and a quantum circuit :math:`U_w` that 
+performs the following: 
+
+.. math::
+  U_w: \ket{x}\ket{q} \to \ket{x}\ket{q\oplus f(x)}
+
+where :math:`f(x)=1` iff :math:`\ket{x}\in\{\ket{y}_i\}`, construct a quantum circuit that when
+given the uniform superposition :math:`\ket{s} = \frac{1}{\sqrt{N}}\sum\limits^{N-1}_{i=0}\ket{x_i}`
+as input, produces a state :math:`\ket{s'}` that, when measured, produces a state :math:`\{y_i\}`
+with probability near one.
+
+As an example, take :math:`U_w: \ket{x}\ket{q} \to \ket{x\cdot\vec{1}}\ket{q\oplus x\cdot\vec{1}}`.
```

Comment:
`\ket{x\cdot\vec{1}}` -> `\ket{x}`? Also what do you mean by `x\cdot\vec{1}`? Is that a bit vector dot product? What about the length of `x` then?

Reply:
It was intended to be the vector of ones of the same dimension, I'll specify. And yes, bitwise dot product.